lib/jxl/enc_heuristics.cc

```diff
@@ -307,9 +309,57 @@ static void DownsampleImage2_Sharper(const ImageF& input, ImageF* output) {
           sum += row_in[ky][ix] * kernel[ky * kernelx + kx];
         }
       }
+
       row_out[x] = sum;
-      if (row_out[x] < min) row_out[x] = min;
-      if (row_out[x] > max) row_out[x] = max;
+
+      const constexpr float mask_strength = 7;
+      float a = std::abs(row_mask[x]) * mask_strength;
```

Comment:
comment here what are the benefits and disadvantages of using clamping and why and where the 'a' variable is increased

Reply:
Done